lib/src/main/java/io/ably/lib/transport/ConnectionManager.java

```diff
@@ -790,7 +797,13 @@ public synchronized State getConnectionState() {
     public synchronized void connect() {
         /* connect() is the only action that will bring the ConnectionManager out of a terminal currentState */
         if(currentState.terminal || currentState.state == ConnectionState.initialized) {
-            startup();
+            try {
+                startup();
+            } catch(InterruptedException e) {
```

Comment:
Which thread exactly throws this exception and which one catches it?

Reply:
I believe external thread is responsible for catching it.

Reply:
Thanks going through this 👍

Reply:
Okay, lgtm. 
Okay, so we are following https://www.baeldung.com/java-interrupted-exception#2-restore-the-interrupt.
The concept is similar to cancellationToken` in dotnet, `cancellationContext` in go .. etc, 
Feels like one of the thread synchronization mechanism.